src/block/block_compact.c

```diff
@@ -443,13 +445,15 @@ __wt_block_compact_skip(WT_SESSION_IMPL *session, WT_BLOCK *block, bool *skipp)
 
     /*
      * We do compaction by copying blocks from the end of the file to the beginning of the file, and
-     * we need some metrics to decide if it's worth doing. Ignore small files, and files where we
-     * are unlikely to recover 10% of the file.
+     * we need some metrics to decide if it's worth doing. Ignore files smaller than the configured
+     * threshold, and files where we are unlikely to recover 10% of the file.
      */
-    if (block->size <= WT_MEGABYTE) {
+    if (block->compact_pages_reviewed == 0 &&
```

Comment:
Should I leave the check against 1MB here? It could go against the threshold if it is set to a value under 1MB.

Reply:
If we set the minimum free space target as 1MB, we can remove this condition here.

Reply:
We are already checking the free space target below, so checking for the free space target here is technically not necessary. But I suppose it wouldn't hurt either.

I'd be however inclined to either keep the original 1 MB check, or if not, at least replace it with the check that ensures that we are not about to compact an empty table. Otherwise we'd need to go through the compact algorithm to ensure that nothing strange happens if someone runs it on an empty file.

Reply:
I agree, we shouldn't go against the threshold here. Instead, setting a minimum makes sense to preserve the current behaviour of "ignoring small files".

Reply:
Preserved behaviour which would ignore small files here: cffc3c54ca6383964a5e11894996cf8d311956a8